unreal/src/context.rs

```diff
@@ -0,0 +1,160 @@
+/// [Source](https://github.com/EpicGames/UnrealEngine/blob/b70f31f6645d764bcb55829228918a6e3b571e0b/Engine/Source/Runtime/Core/Private/GenericPlatform/GenericPlatformCrashContext.cpp)
+pub struct Unreal4Context {
+    pub runtime_properties: Option<Unreal4ContextRuntimeProperties>,
+    pub platform_properties: Option<Unreal4ContextPlatformProperties>,
+}
+
+/// RuntimeProperties context element
+/// [Source](https://github.com/EpicGames/UnrealEngine/blob/b70f31f6645d764bcb55829228918a6e3b571e0b/Engine/Source/Runtime/Core/Private/GenericPlatform/GenericPlatformCrashContext.cpp#L274)
+#[derive(Default, Clone, Debug)]
+pub struct Unreal4ContextRuntimeProperties {
+    /// CrashGUID
+    pub crash_guid: Option<String>,
+    /// ProcessId
+    pub process_id: Option<u32>,
+}
+
+impl Unreal4ContextRuntimeProperties {
+    fn empty() -> Unreal4ContextRuntimeProperties {
+        Unreal4ContextRuntimeProperties {
+            crash_guid: None,
+            process_id: None,
+        }
+    }
+}
+
+/// Platform specific properties.
+/// [Source[(https://github.com/EpicGames/UnrealEngine/blob/b70f31f6645d764bcb55829228918a6e3b571e0b/Engine/Source/Runtime/Core/Private/GenericPlatform/GenericPlatformCrashContext.cpp#L451-L455)
+/// [Windows](https://github.com/EpicGames/UnrealEngine/blob/b70f31f6645d764bcb55829228918a6e3b571e0b/Engine/Source/Runtime/Core/Private/Windows/WindowsPlatformCrashContext.cpp#L39-L44)
+pub struct Unreal4ContextPlatformProperties {
+    /// Whether the crash happened on a Windows device.
+    pub is_windows: Option<bool>,
+    /// Platform-specific UE4 Core value.
+    pub callback_result: Option<i32>,
+}
+
+impl Unreal4ContextPlatformProperties {
+    fn empty() -> Unreal4ContextPlatformProperties {
+        Unreal4ContextPlatformProperties {
+            is_windows: None,
+            callback_result: None,
+        }
+    }
+}
+
+impl Unreal4Context {
+    pub(crate) fn from_crash(crash: &Unreal4Crash) -> Result<Option<Self>, Unreal4Error> {
+        let file = match crash.get_file_slice(Unreal4FileType::Context)? {
+            Some(f) => f,
+            None => return Ok(None),
+        };
+
+        let root = Element::from_reader(file).map_err(|e| Unreal4Error::InvalidXml(e))?;
+
+        Ok(Some(Unreal4Context {
+            runtime_properties: get_runtime_properties(&root),
+            platform_properties: get_platform_properties(&root),
+        }))
+    }
+}
+
+fn get_runtime_properties(root: &Element) -> Option<Unreal4ContextRuntimeProperties> {
+    let list = root.find("RuntimeProperties")?;
+
+    let mut rv = Unreal4ContextRuntimeProperties::empty();
+
+    for child in list.children() {
+        if child.tag() == &QName::from("CrashGUID") {
+            let text = child.text().to_string();
+            if text != "" {
+                rv.crash_guid = Some(child.text().to_string());
+            }
+        } else if child.tag() == &QName::from("ProcessId") {
+            let int = child.text().parse::<u32>();
```

Comment:
This entire block can probably be rewritten as `rv.process_id = child.text().parse::<u32>().ok()`;